Mathlib/LinearAlgebra/Matrix/FixedDetMatrices.lean

```diff
@@ -0,0 +1,57 @@
+/-
+Copyright (c) 2024 Chris Birkbeck. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chris Birkbeck
+-/
+
+import Mathlib.LinearAlgebra.Matrix.SpecialLinearGroup
+
+/-!
+# Matrices with fixed determinant
+
+This file defines the type of matrices with fixed determinant `m` and proves some basic results
+about them.
+
+Note: Some of this was done originally in Lean 3 in the
+kbb (https://github.com/kim-em/kbb/tree/master) repository, so credit to those authors.
+-/
+
+variable (n : Type*) [DecidableEq n] [Fintype n] (R : Type*) [CommRing R]
+
+/--The set of matrices with fixed determinant `m`. -/
+def FixedDetMatrix (m : R) := { A : Matrix n n R // A.det = m }
+
+namespace FixedDetMatrix
+
+open ModularGroup Matrix SpecialLinearGroup MatrixGroups
+
+lemma ext' (m : R) {A B : FixedDetMatrix n R m} (h : A.1 = B.1) : A = B := by
+  cases A; cases B
+  congr
+
+@[ext]
+lemma ext (m : R) {A B : FixedDetMatrix n R m} (h : ∀ i j , A.1 i j = B.1 i j) : A = B := by
```

Comment:
I'm curious if there's a particular reason to make `m` explicit here.

Reply:
Ah good point, fixed!